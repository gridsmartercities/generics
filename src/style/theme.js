const theme = {
  main: {},
  input: {},
  label: {},
  buttons: {},
  containers: {}
};

export default theme;
