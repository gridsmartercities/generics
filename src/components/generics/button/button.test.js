import "jest-dom/extend-expect";
import React from "react";
import { Simulate } from "react-dom/test-utils";
import { render } from "@testing-library/react";
import Button from "./index";
import theme from "../../../style/theme";
import { ThemeProvider } from "styled-components";

test("Regular Button Component", () => {
  const { getByLabelText, container } = render(
    <ThemeProvider theme={theme}>
      <Button>Hello</Button>
    </ThemeProvider>
  );
  const button = document.querySelector("button");
  expect(button).toHaveTextContent("Hello");
  expect(container.firstChild).toMatchSnapshot();
});

test("Link Button Component", () => {
  const { getByLabelText, container } = render(
    <ThemeProvider theme={theme}>
      <Button href="/hello">Hello</Button>
    </ThemeProvider>
  );

  const link = document.querySelector("a");
  expect(link.nodeName).toBe("A");
  expect(link).toHaveTextContent("Hello");
  expect(link.href).toEqual("http://localhost/hello");
  expect(container.firstChild).toMatchSnapshot();
});
