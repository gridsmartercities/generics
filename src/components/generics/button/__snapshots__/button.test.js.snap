// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Button Component 1`] = `
<div
  class="sc-bdVaJa jpqvBK"
>
  <a
    class="sc-bwzfXH cFcRRy"
    href="/hello"
    type="link"
  >
    Hello
  </a>
</div>
`;

exports[`Regular Button Component 1`] = `
<div
  class="sc-bdVaJa jpqvBK"
>
  <button
    class="sc-htpNat iHNvzc"
    type="button"
  >
    Hello
  </button>
</div>
`;
